### PR TITLE
🚚 🐛🔥 Streamline Overall Code (Mostly Back end)

### PR DESCRIPTION
**Pull request by anarmawala**
🚚 Moved values to the proper functions
🚚 Moved Regex to global
🐛 Fix Regex for name
🔥 Removed redundant/vulnerable `console.log` statements
💥 Change Testing port to 8085 to stop conflicting with `mongo-express` and `redis-commander` and `nginx`

### CODE REVIEW

File: api/src/routes/auth/index.ts [hunk 1]
```diff
@@ -171,20 +167,21 @@ router.post('/update', isAuthenticated, async (req: Request, res: Response) => {
...
                 res.status(400).send('Email is invalid')
                 return
             }
         }
 
         // * Update in mongoose
         const updatedUser = await User.findOneAndUpdate({ _id: user._id }, updates, { new: true })
-        res.status(200).send(updatedUser)
+
+        // * Makes sure unnecessary keys/values are not sent
```
Comment (alexchomiak) [inline api/src/routes/auth/index.ts:185]: Delete what you don't want to send, don't specify each thing you are sending. It is less scalable that way. Just incase we want to update other attributes using this route later down the road.

**Review by alexchomiak — CHANGES_REQUESTED** [review]
Please make the change in regards to the comment I made. Otherwise everything looks good! Great job! 👍  🌮

File: api/src/routes/auth/index.ts (hunk 1 above)
Comment (anarmawala) [reply api/src/routes/auth/index.ts:185]: Good point, changed in 1b725afba1c7ed6ffdcbf95c4a202822c0821f20. 🎉

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "cSpell.words": [
```
Comment (bmiddha) [inline .vscode/settings.json]: remove this from project config

File: api/src/tests/auth.test.ts [hunk 3]
```diff
@@ -350,27 +351,27 @@ describe('Authentication Tests', () => {
         it('Can update user correctly', async () => {
             // * Signup
             await client.post('signup', {
-                firstname: 'Bharat',
-                lastname: 'Middha',
-                email: 'achomi2@uic.edu',
-                password: 'oldPassword1$'
+                firstname: 'Jimmy',
+                lastname: 'Falcon',
+                email: 'jimmy@registrum.app',
```
Comment (bmiddha) [inline api/src/tests/auth.test.ts]: use example.com. They were specifically created by IANA to be used as example domains.

File: api/src/tests/auth.test.ts [hunk 4]
```diff
@@ -350,27 +351,27 @@ describe('Authentication Tests', () => {
...
             })
 
             // * Logout
             await client.post('logout')
 
             // * Log back in with new password
             const response = await client.post('/login', {
-                email: 'achomi2@uic.edu',
-                password: 'theRealAlex1$'
+                email: 'jimmy@registrum.app',
```
Comment (bmiddha) [inline api/src/tests/auth.test.ts]: use example.com. They were specifically created by IANA to be used as example domains.

File: api/src/tests/banner.test.ts [hunk 5]
```diff
@@ -69,11 +40,13 @@ describe('Class Tests', () => {
...
+        server = app.listen(PORT)
+
         const response = await client.post(`auth/signup`, {
-            firstname: 'Clark',
-            lastname: 'Chen',
-            email: 'schen237@uic.edu',
-            password: 'theRealClark1$'
+            firstname: 'John',
+            lastname: 'Doe',
+            email: 'example@registrum.app',
```
Comment (bmiddha) [inline api/src/tests/banner.test.ts]: use example.com. They were specifically created by IANA to be used as example domains.

File: api/src/tests/banner.test.ts [hunk 6]
```diff
@@ -90,17 +63,47 @@ describe('Class Tests', () => {
...
+
+        // * Close Server
+        server.close()
+    })
+
     beforeEach(async () => {
         await client.post('auth/login', {
-            email: 'schen237@uic.edu',
-            password: 'theRealClark1$'
+            email: 'example@registrum.app',
```
Comment (bmiddha) [inline api/src/tests/banner.test.ts]: use example.com. They were specifically created by IANA to be used as example domains.

File: api/src/tests/class.test.ts [hunk 7]
```diff
@@ -5,27 +5,42 @@ import { CookieJar } from 'tough-cookie'
...
+    })
+    axiosCookieJarSupport(client)
+
+    beforeAll(async () => {
+        server = app.listen(PORT)
+
+        const response = await client.post(`auth/signup`, {
+            firstname: 'John',
+            lastname: 'Doe',
+            email: 'example@registrum.app',
```
Comment (bmiddha) [inline api/src/tests/class.test.ts]: use example.com. They were specifically created by IANA to be used as example domains.

**Review by bmiddha — CHANGES_REQUESTED** [review]

File: api/src/tests/class.test.ts (hunk 7 above)
Comment (anarmawala) [reply api/src/tests/class.test.ts]: Changed in f3631033ac6f5822cec92443b9135e97b69f5608

File: api/src/tests/banner.test.ts (hunk 6 above)
Comment (anarmawala) [reply api/src/tests/banner.test.ts]: Changed in f3631033ac6f5822cec92443b9135e97b69f5608

File: api/src/tests/auth.test.ts (hunk 3 above)
Comment (anarmawala) [reply api/src/tests/auth.test.ts]: Changed in f3631033ac6f5822cec92443b9135e97b69f5608

File: api/src/tests/banner.test.ts (hunk 5 above)
Comment (anarmawala) [reply api/src/tests/banner.test.ts]: Changed in f3631033ac6f5822cec92443b9135e97b69f5608

File: api/src/tests/auth.test.ts (hunk 4 above)
Comment (anarmawala) [reply api/src/tests/auth.test.ts]: Changed in f3631033ac6f5822cec92443b9135e97b69f5608

File: .vscode/settings.json (hunk 2 above)
Comment (anarmawala) [reply .vscode/settings.json]: Removed in 7255e0681e1009505f01beeedf19d0b169da3da9

**Review by bmiddha — APPROVED** [review]
Looks good to me! 🎉 
🌮 Taco Tuesday! 🌮

**Review by alexchomiak — APPROVED** [review]

**Comment by alexchomiak** [comment]
Thanks for making those changes! Everything looks good! 🎉 🌮

**Pull request merged**